### PR TITLE
PHOENIX-6881 Implement the applicable Date/Time features from JDBC 4.2

### PR DESCRIPTION
**Pull request by stoty**

### CODE REVIEW

**Review by richardantal — APPROVED** [review]
LGTM

File: phoenix-core/src/main/java/org/apache/phoenix/jdbc/PhoenixPreparedStatement.java [hunk 1]
```diff
@@ -528,16 +519,20 @@ public void setTimestamp(int parameterIndex, Timestamp x) throws SQLException {
...
+
+    private java.sql.Timestamp processTimestamp(Timestamp x, Calendar cal) {
         if (x != null) {
             if (connection.isApplyTimeZoneDisplacement()) {
-                x = DateUtil.applyInputDisplacement(x, cal.getTimeZone());
+                return DateUtil.applyInputDisplacement(x, cal.getTimeZone());
             } else {
                 int nanos = x.getNanos();
                 x = new Timestamp(x.getTime());
                 x.setNanos(nanos);
```
[the remark this answers is not in the document]
Comment (stoty) [reply phoenix-core/src/main/java/org/apache/phoenix/jdbc/PhoenixPreparedStatement.java:532]: Yes, handling nanos is awkward with java.sql.Timestamp.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/GetSetObjectIT.java [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+
+            ResultSet rs = stmt.executeQuery("select * from " + tableName);
+            while (rs.next()) {
+                assertArrayEquals(bytes, rs.getBytes(2));
+                assertArrayEquals(bytes, (byte[]) rs.getObject(2));
+                assertArrayEquals(bytes, rs.getObject(2, byte[].class));
+                try {
+                    rs.getObject(2, Long.class);
+                    fail("We only implement byte[]");
+                } catch (SQLException e) {
```
[the remark this answers is not in the document]
Comment (stoty) [reply phoenix-core/src/it/java/org/apache/phoenix/end2end/GetSetObjectIT.java:82]: Makes sense.

**Comment by stoty** [comment]
Thank you, I have implemented your suggestion in the test before commit.

**Pull request closed without merging**